src/components/body/content/categories/CategoriesAdmin.tsx

```diff
@@ -1,48 +1,54 @@
 import {
   MouseEvent, useEffect, useState
 } from "react";
-import {archiveCategory, getCategory} from "../../../../store/categorySlice/categorySlice";
+import {
+  archiveCategory, deleteCategory, getCategory
+} from "../../../../store/categorySlice/categorySlice";
 import {useAppDispatch, useAppSelector} from "../../../../store/hooks";
 import Add from "../../../icons/add.svg";
 import Delete from "../../../icons/delete.svg";
 import Edit from "../../../icons/edit.svg";
-import IT from "../../../icons/examplImage/IT.svg";
 import Loader from "../../../Loader";
+import ModalCategory from "./actions/ModalCategory";
+import {useIntl} from "react-intl";
 import {useNavigate} from "react-router-dom";
 
-const CategoriesAdmin=()=>{
-  const [isGetCategory, setIsGetCategory] = useState(false);
+const CategoriesAdmin = () => {
+  const [isClose, setIsClose] = useState(false);
+  const [idCategory, setIdCategory]=useState(0);
+  const intl = useIntl();
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
   const allCategories = useAppSelector((state) => state.categories.categories);
   const loading = useAppSelector((state) => state.categories.loading);
 
-  useEffect(()=>{
+  useEffect(() => {
     dispatch(getCategory());
-  },[dispatch]);
+  }, [dispatch]);
 
   const handleEdit = (event: MouseEvent) => {
     navigate(`/categories/updateCategory/${event.currentTarget.id}`);
   };
 
-  const handleDelete = async (category:
-    { id: number, name: string, description: string, status: string }
-  ) => {
-    await dispatch(archiveCategory(category));
-    dispatch(getCategory());
-    setIsGetCategory(!isGetCategory);
+  const handleDelete = async (category: {
+    id: number, name: string, description: string, status: string, amount_lessons: number
+  }) => {
+    setIdCategory(category.id);
+    category.amount_lessons > 0 ?
```

Comment:
Please, use ternary operator only with return or assign. Do not use it instead of if..else - bad practice.


Read here: https://learn.javascript.ru/ifelse#netraditsionnoe-ispolzovanie